src/EventSubscriber/GeneratedUserIngestControllerSubscriber.php

```diff
@@ -0,0 +1,100 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\EventSubscriber;
+
+use OAT\SimpleRoster\Events\LineItemUpdated;
+use OAT\SimpleRoster\Lti\Service\ColumnGroupResolver;
+use OAT\SimpleRoster\Lti\Service\GenerateGroupIdsService;
+use OAT\SimpleRoster\Repository\Criteria\FindLineItemCriteria;
+use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\Repository\LtiInstanceRepository;
+use OAT\SimpleRoster\Service\Bulk\BulkCreateUsersService;
+use OAT\SimpleRoster\WebHook\UpdateLineItemDto;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class GeneratedUserIngestControllerSubscriber implements EventSubscriberInterface
+{
+    public const NAME = 'line-item.updated';
+
+    private LoggerInterface $logger;
+    private BulkCreateUsersService $createService;
+    private GenerateGroupIdsService $generateGroupIdsService;
+    private LtiInstanceRepository $ltiInstanceRepository;
+    private LineItemRepository $lineItemRepository;
+
+    private bool $enabled;
+
+    public function __construct(
+        LoggerInterface $logger,
+        BulkCreateUsersService $createService,
+        GenerateGroupIdsService $generateGroupIdsService,
+        LtiInstanceRepository $ltiInstanceRepository,
+        LineItemRepository $lineItemRepository,
+        bool $enabled
+    ) {
+        $this->logger = $logger;
+        $this->enabled = $enabled;
+        $this->createService = $createService;
+        $this->generateGroupIdsService = $generateGroupIdsService;
+        $this->ltiInstanceRepository = $ltiInstanceRepository;
+        $this->lineItemRepository = $lineItemRepository;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            LineItemUpdated::NAME => ['onLineItemUpdated', 10],
+        ];
+    }
+
+    public function onLineItemUpdated(LineItemUpdated $event): void
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $this->logger->info("Got LineItemUpdate event", [
+            'line_items_slugs' => $event->getUpdateLineItemCollection()->map(fn($dto) => $dto->getSlug())
+        ]);
+
+        $acceptedSlugs = $event->getUpdateLineItemCollection()
+            ->filter(fn($dto) => $dto->getStatus() === UpdateLineItemDto::STATUS_ACCEPTED)
+            ->map(fn($dto) => $dto->getSlug());
+
+        $batchSize = 20;
+        $ltiCollection = $this->ltiInstanceRepository->findAllAsCollection();
+
+        $this->createService->generate(
+            $this->lineItemRepository->findLineItemsByCriteria(
+                (new FindLineItemCriteria())->addLineItemSlugs(...$acceptedSlugs)
+            )->jsonSerialize(),
+            ['QA1', 'QA2'],
+            $batchSize,
+            new ColumnGroupResolver(
```

Comment:
I am not sure if it is a good practice co instantiate classes like this. it is difficult to mock. would you consider using a factory?

Reply:
You'r right. For now i think it's too much for the first iteration.